axon-framework/testing/commands-events.md

```diff
@@ -167,26 +167,79 @@ During the configuration phase \(i.e. before the first "given" is provided\), yo
 
 ### Test Execution Phase
 
-The execution phase allows you two entry points towards the [validation phase](commands-events.md#validation-phase). Firstly, you can provide a Command to be executed against the command handling component. Similar as with the given Events, if the provided Command is of type `CommandMessage` it will be dispatched as is. The behavior of the invoked handler \(either on the aggregate or as an external handler\) is monitored and compared to the expectations registered in the validation phase.
-
-Secondly, it is possible to elapse a certain time span with the `whenThenTimeElapses(Duration)` and `whenThenTimeAdvancesTo(Instant)` handles. These support testing the publication of `DeadlineMessages`, as is further defined in [this](../deadlines/) chapter.
-
-Note that only activities that occur during the test _execution_ phase are monitored. Any Events or side-effects generated during the "given" phase are not considered in the validation phase.
+The execution phase, or "when phase," allows you several entry points towards the [validation phase](commands-events.md#validation-phase).
+Note that the test fixture only monitors activities that occur during the test _execution_ phase.
+The fixture does not consider any events or side effects generated during the [given phase](#given-phase) in the validation phase.
 
 > **Illegal State Change Detection**
 >
-> During the execution of the test, Axon attempts to detect any illegal state changes in the aggregate under test. It does so by comparing the state of the aggregate after the command execution to the state of the aggregate if it sourced from all "given" and stored events. If that state is not identical, this means that a state change has occurred outside of an aggregate's event handler method. Static and transient fields are ignored in the comparison, as they typically contain references to resources.
+> During the execution of the test, Axon attempts to detect any illegal state changes in the aggregate under test.
+> It does so by comparing the aggregate's state after command execution to the aggregate's state when event sourced from all given and stored events.
+> If that state is not identical, a state change occurred outside an aggregate's event handler method.
+> Static and transient fields are ignored in the comparison, as they typically contain references to resources.
 >
 > You can switch detection in the configuration of the fixture with the `setReportIllegalStateChange()` method.
 
+We can separate the execution phase options into roughly three variants:
+
+1. Execute a command,
+2. time progresses, and
+3. invoking a method on the aggregate.
+
+Below is a list of all the operations you can use in the execution phase:
+
+* `when(Object)`:
+  Using the `when` method, you can provide a command for the fixture to execute against the command handling component under test.
```

Comment:
```suggestion
  Using the `when` method, you can provide a command for the fixture to execute against the aggregate under test.
```

Reply:
I think this applies to the other points as well, since it's the `AggregateTestFixture`. I don't believe you can test 
regular command handlers with it

Reply:
So, this is not 100% the case, but thus warrants some form of notice. If you have an external command handler that invokes the repository of the Aggregate under test, you would also validate behavior of that external command handler. Hence, I named this "command handling component" i.o. aggregate. Granted, it's perhaps a little unclear now. I'll conjure up something more to the point.

Reply:
Then again...the core is aggregate testing (hence why it's called the `AggregateTestFixture`, of course), not "ExternalCommandHandlerInvokingAggregatesTestFixture."